src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -57,15 +58,24 @@ class BasePaymentsPage extends React.Component {
         this.hidePasswordPrompt = this.hidePasswordPrompt.bind(this);
         this.navigateToTransferBalancePage = this.navigateToTransferBalancePage.bind(this);
         this.setMenuPosition = this.setMenuPosition.bind(this);
+        this.fetchData = this.fetchData.bind(this);
     }
 
     componentDidMount() {
-        PaymentMethods.getPaymentMethods();
+        this.fetchData();
         if (this.props.shouldListenForResize) {
             this.dimensionsSubscription = Dimensions.addEventListener('change', this.setMenuPosition);
         }
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.network.isOffline === this.props.network.isOffline) {
+            return;
+        }
```

Comment:
So, this will cause us to `fetchData` when we go from offline to online, which makes sense...
previousOffline = `true`
currentOffline = `false`
--> evaluates to `false` and we `fetchData`

But it will also cause us to `fetchData` when we go from online to offline, which makes less sense to me.
previousOffline = `false`
currentOffline = `true`
--> evaluates to `false` and we `fetchData`

Should this conditional be:
```js
if (prevProps.network.isOffline && !this.props.network.isOffline) {
```